tools/item_db_formatter.py

```diff
@@ -0,0 +1,322 @@
+        (0, 'IT_HEALING'),
+        (2, 'IT_USABLE'),
+        (3, 'IT_ETC'),
+        (4, 'IT_WEAPON'),
+        (5, 'IT_ARMOR'),
+        (6, 'IT_CARD'),
+        (7, 'IT_PETEGG'),
+        (8, 'IT_PETARMOR'),
+        (10, 'IT_AMMO'),
+        (11, 'IT_DELAYCONSUME'),
+        (18, 'IT_CASH')
+    )
+    for type_ in types:
+        if item_type in type_:
+            return type_[1]
+    return ''
+
+
+def item_get_job_const(job: int):
+    job_names = (
+        "Novice",
+        "Swordsman",
+        "Magician",
+        "Archer",
+        "Acolyte",
+        "Merchant",
+        "Thief",
+        "Knight",
+        "Priest",
+        "Wizard",
+        "Blacksmith",
+        "Hunter",
+        "Assassin",
+        "Unused",
+        "Crusader",
+        "Monk",
+        "Sage",
+        "Rogue",
+        "Alchemist",
+        "Bard",
+        "Unused",
+        "Taekwon",
+        "Star_Gladiator",
+        "Soul_Linker",
+        "Gunslinger",
+        "Ninja",
+        "Gangsi",
+        "Death_Knight",
+        "Dark_Collector",
+        "Kagerou",
+        "Rebellion",
+        "Summoner"
+    )
+    job_mask = int(hex(job), 16)
+    result = ""
+    if job_mask == '':
+        return "\n"
+
+    if job_mask & ALL_JOBS == ALL_JOBS:
+        return "\t\tAll: true\n"
+    if job_mask & ALL_EXCEPT_NOVICE == ALL_EXCEPT_NOVICE:
+        result = "\t\tAll: true\n"
+        result = result + "\t\tNovice: false\n"
+        return result
+
+    for x, job in enumerate(job_names):
+        current_bit = 1 << x
+        if job_mask & current_bit == current_bit:
+            result += '\t\t{job}: true\n'.format(job=job_names[x])
+    return result
+
+
+def item_get_trade_options(trade):
+    if not trade:
+        return ''
+    result = ''
+    for trade_opt in trade.items():
+        result += '\t\t{key}: {value}\n'.format(key=trade_opt[0], value=str(trade_opt[1]).lower())
+    return result
+
+
+def item_get_nouse_options(nouse):
+    if not nouse:
+        return ''
+
+    result = ''
+    for nouse_opt in nouse.items():
+        result += '\t\t{key}: {value}\n'.format(key=nouse_opt[0], value=str(nouse_opt[1]).lower())
+    return result
+
+
+def get_optional_fields(item):
+    # Check all fields
+    # if no old field found just don't put it in new file
+    optional_fields = ''
+    item_type = item_type_formatter(item.get('Type'))
+    buy = item.get('Buy')
+    sell = item.get('Sell')
+    weight = item.get('Weight')
+    atk = item.get('Atk')
+    matk = item.get('Matk')
+    defense = item.get('Def')
+    range_ = item.get('Range')
+    slots = item.get('Slots')
+    job = item_get_job_const(item.get('Job', 0))
+    upper = item_get_upper_mask(item.get('Upper', None))
+    gender = item_get_gender(item.get('Gender'))
+    loc = item.get('Loc')
+    weapon_lv = item.get('WeaponLv')
+    equip_lv = item.get('EquipLv')
+    refine = item.get('Refine')
+    disable_options = item.get('DisableOptions')
+    subtype = item.get('Subtype')
```

Comment:
since this is suppose to convert from the old format to the new one, `Subtype` and `ViewSprite` should be added to the item by reading `View` and `Type`, if it's a weapon or ammo then print it as `Subtype` else print it as `ViewSprite`, for reference check a98b2d93, a31f5f98, 513ccda4